### PR TITLE
Fix receivedAt to use iso date format as input and store as date - Closes #1156

### LINKED ISSUE #1156: Transaction receiveAt doesn't match the input/output
### Expected behavior
transaction.toJSON() should return instanciatable object.

### Actual behavior
toJSON() returns string, but constructor expects Data object

### Which version(s) does this affect? (Environment, OS, etc...)
2.0-rc.2-

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
property receivedAt did not have matching property input/output

### How did I fix it?
`receivedAt` takes in valid string format of date, and outputs ISODateFormat.
internally it stores Date object

### How to test it?
`npm t`

### Review checklist

* The PR resolves #1156
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Review by pablitovicente — DISMISSED** [review]

File: packages/lisk-transactions/src/base_transaction.ts [hunk 1]
```diff
@@ -162,7 +156,9 @@ export abstract class BaseTransaction {
 		this._signSignature = rawTransaction.signSignature;
 		this.timestamp = rawTransaction.timestamp;
 		this.type = rawTransaction.type;
-		this.receivedAt = rawTransaction.receivedAt || new Date();
```
Comment (SargeKhan) [inline packages/lisk-transactions/src/base_transaction.ts:165]: `receivedAt` is a property which is only maintained in memory and is allowed to be undefined. Transactions saved in the blockchain don't maintain this property and every node maintains it's own value for `receivedAt` property. Therefore, I think it should only be explicitly set, if the `receivedAt` property isn't provided to the constructor, it should be undefined/null.

**Review by SargeKhan — CHANGES_REQUESTED** [review]

File: packages/lisk-transactions/src/base_transaction.ts (hunk 1 above)
Comment (shuse2) [reply packages/lisk-transactions/src/base_transaction.ts:165]: If we allow `receivedAt` as undefined, how should we calculate `timeElapsed` in the `isExpired` function?
it needs current time and the created time

Comment (SargeKhan) [reply packages/lisk-transactions/src/base_transaction.ts:165]: I guess receivedAt is only used in the transaction pool. Therefore, it's the responsibility of the transaction pool to assign receivedAt property for each transaction as soon as it enters transaction pool. It's done here: https://github.com/LiskHQ/lisk-elements/blob/development/packages/lisk-transaction-pool/src/transaction_pool.ts#L439

Comment (shuse2) [reply packages/lisk-transactions/src/base_transaction.ts:165]: so if `isExpired` is called without the receivedAt, it should throw an error?

**Review by pablitovicente — APPROVED** [review]

**Review by SargeKhan — APPROVED** [review]

**Pull request merged**